### PR TITLE
Mopac

### PR DESCRIPTION
**Pull request by dgasmith**
This PR begins a MOPAC Harness thanks help from @godotalgorithm. Not too much remaining besides some additional testing, should also add the standard files to the QCEngineRecords repo so that we can test offline.

I really need to get on the custom MolSSI CI server so that we can run through all of these programs.

### CODE REVIEW

File: qcengine/programs/mopac.py [hunk 1]
```diff
@@ -0,0 +1,256 @@
+"""
+Calls the Psi4 executable.
```
Comment (Lnaden) [inline qcengine/programs/mopac.py]: Copy typo

File: qcengine/programs/mopac.py [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+
+    def get_version(self) -> str:
+        self.found(raise_error=True)
+
+        # Not really possible to pull at the moment, MolSSI will add a version ability
+        return "2016"
+
+    def compute(self, input_model: 'ResultInput', config: 'JobConfig') -> 'Result':
+        """
+        Runs Psi4 in API mode
```
Comment (Lnaden) [inline qcengine/programs/mopac.py]: Copy typo

File: qcengine/programs/mopac.py [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+        return which('mopac',
+                     return_bool=True,
+                     raise_error=raise_error,
+                     raise_msg='Please install via http://openmopac.net.')
+
+    def get_version(self) -> str:
+        self.found(raise_error=True)
+
+        # Not really possible to pull at the moment, MolSSI will add a version ability
+        return "2016"
```
[the remark this answers is not in the document]
Comment (dgasmith) [reply qcengine/programs/mopac.py]: It does, but not possible to grab upfront. We will be changing this in MOPAC itself at a later date.

File: qcengine/programs/mopac.py [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+                    template: Optional[str] = None) -> Dict[str, Any]:
+
+        if template is not None:
+            raise KeyError("MOPAC does not currently support input templates.")
+
+        method = input_model.model.method.lower()
+        if method not in {
+                "mndo", "am1", "pm3", "rm1", "mndod", "pm6", "pm6-d3", "pm6-dh+", "pm6-dh2", "pm6-dh2x", "pm6-d3h4",
+                "pm6-3dh4x", "pm7", "pm7-ts"
+        }:
```
[the remark this answers is not in the document]
Comment (dgasmith) [reply qcengine/programs/mopac.py]: Why not a set?

File: qcengine/programs/mopac.py [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+            input_file.append(
+                f"{mol.symbols[x]}  {mol.geometry[x, 0]:17.12f}  {mol.geometry[x, 1]:17.12f}  {mol.geometry[x, 2]:17.12f}"
+            )
+
+        env = os.environ.copy()
+        env["MKL_NUM_THREADS"] = str(config.ncores)
+        env["OMP_NUM_THREADS"] = str(config.ncores)
+
+        return {
+            "commands": ["mopac", "dispatch.mop"],
```
[the remark this answers is not in the document]
Comment (dgasmith) [reply qcengine/programs/mopac.py]: Sure, can you break this into an issue to overhaul everything?

File: qcengine/programs/tests/test_molpro.py [hunk 6]
```diff
@@ -29,7 +29,7 @@ def test_molpro_output_parser(test_case):
     output_ref.pop("provenance", None)
 
     # TODO add `skip` to compare_recursive
-    check = compare_recursive(output_ref, output)
+    check = compare_recursive(output_ref, output, forgive={"stdout"})
```
[the remark this answers is not in the document]
Comment (dgasmith) [reply qcengine/programs/tests/test_molpro.py]: Because it must be a set?

[the remark this answers is not in the document]
Comment (loriab) [reply qcengine/programs/tests/test_molpro.py]: https://github.com/MolSSI/QCElemental/blob/22a7c9b91a7cfeedae61beb47e3a2c3faad0cec9/qcelemental/testing.py#L355-L358

File: qcengine/programs/mopac.py [hunk 7]
```diff
@@ -0,0 +1,256 @@
...
+
+        gradient = data.pop("gradients")
+
+        output = input_model.dict()
+        output["provenance"] = {"creator": "mopac", "version": data.pop("mopac_version")}
+
+        output["properties"] = {}
+        output["properties"]["return_energy"] = data["heat_of_formation"]
+
+        output["extras"] = data
```
[the remark this answers is not in the document]
Comment (dgasmith) [reply qcengine/programs/mopac.py]: We could not sure there is really a standard here however.

**Pull request merged**